gems/pending/disk/modules/miq_disk_cache.rb

```diff
@@ -0,0 +1,109 @@
+require 'rufus/lru'
+require_relative "../MiqDisk"
+require 'ostruct'
+
+module MiqDiskCache
+  MIN_SECTORS_TO_CACHE = 32
+  DEF_CACHE_SIZE       = 100
+  DEBUG_CACHE_STATS    = false
+
+  attr_reader :d_size, :blockSize, :cache_size, :sectors_to_cache, :downstream
+
+  def self.new(downstream, d_info = nil)
```

Comment:
Since this is a module and not a class, prefer a different name for the factory method, otherwise this will create confusing.  Perhaps `.create`?